posthog/approvals/decorators.py

```
@@ -0,0 +1,467 @@
+
+    if result.action == "error":
+        return Response({"error": result.error_message}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    if result.action == "duplicate" and result.change_request:
+        return Response(
+            {
+                "code": "change_request_pending",
+                "error": "A pending approval request already exists for this action",
+                "resource_type": result.resource_type,
+                "resource_id": result.resource_id,
+                "change_request_id": str(result.change_request.id),
+                "existing_change_request": ChangeRequestSerializer(result.change_request).data,
+            },
+            status=status.HTTP_409_CONFLICT,
+        )
+
+    if result.action == "require_approval" and result.change_request:
+        return Response(
+            {
+                "code": "approval_required",
+                "status": "approval_required",
+                "message": result.decision.message if result.decision else "Approval required",
+                "resource_type": result.resource_type,
+                "resource_id": result.resource_id,
+                "change_request_id": str(result.change_request.id),
+                "change_request": ChangeRequestSerializer(result.change_request).data,
+                "required_approvers": result.approvers,
+            },
+            status=status.HTTP_409_CONFLICT,
+        )
+
+    return None
+
+
+def _resolve_actions(action_refs: Union[type, str, list]) -> list:
+    """Resolve action reference(s) to a list of action classes."""
+    if isinstance(action_refs, list):
+        refs = action_refs
+    else:
+        refs = [action_refs]
+
+    actions = []
+    for ref in refs:
+        if isinstance(ref, str):
+            action_class = get_action(ref)
+            if action_class:
+                actions.append(action_class)
+            else:
+                logger.exception("Failed to resolve action reference: %s", ref)
```

nit: would it be better to raise an error so we can see it in error tracking and alerts?